cmd/ooniprobe/internal/cli/list/list.go

```diff
@@ -19,7 +19,7 @@ func init() {
 			return err
 		}
 		if *resultID > 0 {
-			measurements, err := probeCLI.DB().ListMeasurements(*resultID)
+			measurements, err := probeCLI.ReadDB().ListMeasurements(*resultID)
```

Comment:
This change seems unnecessary to me. In general, I think the whole diff contains too many changes and we can get away with a much smaller changeset. I would recommend to do the following:

1. the `*database.database` type remains `*database.Database` as it was before;
2. all the code in `cmd/ooniprobe` continues to use a `*database.Database` as it did before.

As such, this specific line of code should continue to be `probeCLI.DB`.

If there are exceptions to these general suggestions, I'll let you know.